cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -996,20 +1009,29 @@ private int moveLeadershipInBatch() {
       int numLeadershipToMove = leadershipMovementTasks.size();
       LOG.debug("Executing {} leadership movements in a batch.", numLeadershipToMove);
       // Execute the leadership movements.
-      if (!leadershipMovementTasks.isEmpty() && !_stopRequested.get()) {
+      if (!leadershipMovementTasks.isEmpty() && _stopSignal.get() == NO_STOP_EXECUTION) {
         // Mark leadership movements in progress.
         _executionTaskManager.markTasksInProgress(leadershipMovementTasks);
 
         // Run preferred leader election.
         ExecutorUtils.executePreferredLeaderElection(_kafkaZkClient, leadershipMovementTasks);
         LOG.trace("Waiting for leadership movement batch to finish.");
-        while (!_executionTaskManager.inExecutionTasks().isEmpty() && !_stopRequested.get()) {
+        while (!_executionTaskManager.inExecutionTasks().isEmpty() && _stopSignal.get() == NO_STOP_EXECUTION) {
           waitForExecutionTaskToFinish();
         }
       }
       return numLeadershipToMove;
     }
 
+    private void deleteZnodesToStopExecution() {
+      // Delete znode of ongoing replica movement tasks.
+      _kafkaZkClient.deletePartitionReassignment();
```

Comment:
Before deleting the znode, let's retrieve the stuck partitions and return them to the user in the response of force stop request (we should also log them). So that the user would know which topics are potentially inconsistent wrt RF of their partitions.

-- this also applies to `getControllerId()`.

Reply:
Returning these information in response will be hard.  
Since the real proposal execution happens asynchronously, i.e in the request handling logic, we just set the flag, The `ProposalExecutionRunnable` running in `ProposalExecutor` thread may pick it up [a little bit later](https://github.com/linkedin/cruise-control/blob/ba0c60ce531068f6c5f8192ff9a4ec1b70201d64/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java#L1044). 
Adding log is a good idea.